src/NHibernate/Impl/MultipleQueriesCacheAssembler.cs

```diff
@@ -28,13 +29,29 @@ public object Disassemble(object value, ISessionImplementor session, object owne
 				var singleQueryCached = new List<object>();
 				foreach (object objToCache in itemList)
 				{
-					if (assemblers.Length == 1)
-					{
-						singleQueryCached.Add(assemblers[0].Disassemble(objToCache, session, owner));
-					}
-					else
+					if (objToCache != null)
```

Comment:
If #1783 is merged, this fix will be fixing an obsoleted class.

The future case is maybe already no more failing or failing in another way, since it does no more use multi-query/multi-criteria. (See #1718.)